pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -835,6 +835,13 @@ func (m *kubeGenericRuntimeManager) GetPodStatus(uid kubetypes.UID, name, namesp
 	})
 	glog.V(4).Infof("getSandboxIDByPodUID got sandbox IDs %q for pod %q", podSandboxIDs, podFullName)
 
+	// Get statuses of all containers visible in the pod.
```

Comment:
I'd like to minimize changes to non-dockershim packages, and the non-windows code flow if possible. The assumption here is also too fragile. Let's try coming up with a smaller, more isolated method.

@karataliu what exactly do we expect dockershim to do for the podsandbox/pause/infra container on windows? I don't see any code preventing dockershim (or even the old dockertools) to try creating such a container. Do we just create this container even though it's meaningless for windows right now?

When calling PodSandboxStatus, is it possible to add a function for windows to simply retrieve and inspect the user container associated with the sandbox? This way we simply get the correct IP for the pod without having to cache it at all. We can use label (sandbox ID) or name parsing (as you did in this PR) to find the correct container. WDYT?